### DISCUSSION TITLE
mlflow R package: Use stages despite deprication?

### DISCUSSION THREAD

**Discussion by chamaoskurumi** [mlflow/mlflow#12668] [Q&A] [answered]
* The latest [mlflow R Package](https://cran.r-project.org/web/packages/mlflow/mlflow.pdf) (Version 2.13.1) does not include any methods that allow setting aliases.
* Whereas in Python one can use `client.set_registered_model_alias()` or `client.delete_registered_model_alias()`, the R package only provides [`mlflow::mlflow-transition-model-version-stage`](https://mlflow.org/docs/latest/R-api.html#mlflow-transition-model-version-stage) at the moment.
* The [concept of stages will be depricated](https://mlflow.org/docs/latest/model-registry.html#deprecated-using-model-stages), best practice would obiviously be to use aliases...

Questions:

* How should I deal with this situation, should I just use stages until the R package gets updated? What would be the most sensible approach here? Using tags and accepting the problem that tags are not unique?
* Does anyone know if the alias related methods will be introduced into the R package anytime soon?

**Comment by B-Step62** [comment, marked as answer]
@chamaoskurumi 

Yes, the R SDK does not support model alias yet. It is only supported in the Python version for now, and we may not be able to add the compatibility soon within limited development resource.

I would recommend continue using model stages for now. We will not likely shut down the feature very soon e.g. next a few minor versions. When we actually remove it, we will make sure there is a pathway for R/Java users to continue using MLflow Model Registry.

Lastly, contribution is always more than welcome! The model registry client implementation should be straight-forward, given that the backend is already there. Also please feel free to [file a feature request](https://github.com/mlflow/mlflow/issues/new?assignees=&labels=enhancement&projects=&template=feature_request_template.yaml&title=%5BFR%5D) in the repository🙂

**Comment by chamaoskurumi** [comment]
I found a somewhat okeyish workaround in R: I just use the **mlflow rest api** (https://mlflow.org/docs/latest/rest-api.html#set-registered-model-alias) from R to set the alias...


    mlflowSetRegisteredModelAlias <- function(mlflowUri, modelName, alias, version) {
      # Define api request payload
      payload <- list(name = modelName,
                      alias = alias,
                      version = as.character(version))
      
      # Make the POST request
      mlflowApiEndpoint <- paste0(mlflowUri, "api/2.0/mlflow/registered-models/alias")
      response <- mlflowApiEndpoint %>%
        httr2::request() %>%
        httr2::req_body_json(payload) %>%
        httr2::req_perform()
    }

For deleting an alias it would be with ad DELETE method:

 

     payload <- list(name = modelName, alias = alias)
      
      # Make the DELETE request
      response <-  mlflowApiEndpoint %>%
        httr2::request() %>%
        httr2::req_method("DELETE") %>%
        httr2::req_body_json(payload) %>%
        httr2::req_perform()
